sklearn/utils/tests/test_multiclass.py

```diff
@@ -296,15 +298,44 @@ def test_is_multilabel():
                 for exmpl_sparse in examples_sparse:
                     assert sparse_exp == is_multilabel(
                         exmpl_sparse
-                    ), "is_multilabel(%r) should be %s" % (exmpl_sparse, sparse_exp)
+                    ), f"is_multilabel({exmpl_sparse!r}) should be {sparse_exp}"
 
             # Densify sparse examples before testing
             if issparse(example):
                 example = example.toarray()
 
             assert dense_exp == is_multilabel(
                 example
-            ), "is_multilabel(%r) should be %s" % (example, dense_exp)
+            ), f"is_multilabel({example!r}) should be {dense_exp}"
+
+
+@pytest.mark.parametrize(
+    "array_namespace, device, dtype",
+    yield_namespace_device_dtype_combinations(),
+)
+def test_is_multilabel_array_api_compliance(array_namespace, device, dtype):
+    xp, device, dtype = _array_api_for_tests(array_namespace, device, dtype)
+
+    for group, group_examples in EXAMPLES.items():
+        dense_exp = group == "multilabel-indicator"
+        for example in group_examples:
+            if not (
+                hasattr(example, "__array__")
+                and np.asarray(example).dtype.kind in "biuf"
+            ):
+                continue
+            # Densify sparse examples before testing
+            if issparse(example):
+                example = example.toarray()
+            if np.asarray(example).dtype.kind == "f":
+                example = np.asarray(example, dtype=dtype)
+            else:
+                example = np.asarray(example)
+            example = xp.asarray(example, device=device)
+
+            assert dense_exp == is_multilabel(
+                example
+            ), f"is_multilabel({example!r}) should be {dense_exp}"
 
```

Comment:
We have to explicitly enable Array API support, otherwise `get_namespace` always returns a Numpy wrapper.

```suggestion
            with config_context(array_api_dispatch=True):
                assert dense_exp == is_multilabel(
                    example
                ), f"is_multilabel({example!r}) should be {dense_exp}"

```

Reply:
Thanks for notice.
This function stop working
```python
def _is_integral_float(y):
    xp, is_array_api_compliant = get_namespace(y)
    return xp.isdtype(y.dtype, "real floating") and bool(xp.all(xp.astype(y, int) == y))
```

Because it can not promote xp.astype(y, int) to float for comparison.
`bool(xp.all(xp.astype((xp.astype(y, xp.int64)), y.dtype) == y))`
This fix passes tests, but do you think there is a better solution?